pkg/3scale/amp/operator/backend_options_provider.go

```diff
@@ -0,0 +1,144 @@
+package operator
+
+import (
+	"fmt"
+
+	"github.com/3scale/3scale-operator/pkg/3scale/amp/component"
+	appsv1alpha1 "github.com/3scale/3scale-operator/pkg/apis/apps/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type OperatorBackendOptionsProvider struct {
+	apimanagerSpec *appsv1alpha1.APIManagerSpec
+	namespace      string
+	client         client.Client
+	backendOptions *component.BackendOptions
+	secretSource   *SecretSource
+}
+
+func NewOperatorBackendOptionsProvider(apimanagerSpec *appsv1alpha1.APIManagerSpec, namespace string, client client.Client) *OperatorBackendOptionsProvider {
+	return &OperatorBackendOptionsProvider{
+		apimanagerSpec: apimanagerSpec,
+		namespace:      namespace,
+		client:         client,
+		backendOptions: component.NewBackendOptions(),
+		secretSource:   NewSecretSource(client, namespace),
+	}
+}
+
+func (o *OperatorBackendOptionsProvider) GetBackendOptions() (*component.BackendOptions, error) {
+	o.backendOptions.AppLabel = *o.apimanagerSpec.AppLabel
+	o.backendOptions.TenantName = *o.apimanagerSpec.TenantName
+	o.backendOptions.WildcardDomain = o.apimanagerSpec.WildcardDomain
+
+	err := o.setSecretBasedOptions()
+	if err != nil {
+		return nil, err
+	}
+
+	o.setResourceRequirementsOptions()
+	o.setReplicas()
+
+	err = o.backendOptions.Validate()
+	return o.backendOptions, err
+}
+
+func (o *OperatorBackendOptionsProvider) setSecretBasedOptions() error {
```

Comment:
Do you think it might be a good idea to modularize the setSecretBasedOptions method? Like separating the generation of cases into different methods so we group related sets of options to make them easily localizable too. For example have a method for the options related to system-backend secret, other for the backend-api secret etc...

It seems large and a little bit hard to find compared with when we had functions

Reply:
Yes, can be an improvement.

Reply:
tried to simplify but result is not good enough. Improvement for future PR's